swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -16,61 +16,44 @@
 import Workflow
 import WorkflowUI
 
-public final class SplitScreenContainerViewController<LeadingScreenType: Screen, TrailingScreenType: Screen>: ScreenViewController<SplitScreenContainerViewController.ContainerScreen> {
-    public typealias ContainerScreen = SplitScreenContainerScreen<LeadingScreenType, TrailingScreenType>
+internal final class SplitScreenContainerViewController<LeadingScreenType: Screen, TrailingScreenType: Screen>: ScreenViewController<SplitScreenContainerViewController.ContainerScreen> {
+    internal typealias ContainerScreen = SplitScreenContainerScreen<LeadingScreenType, TrailingScreenType>
     
-    private var leadingContentViewController: ScreenViewController<LeadingScreenType>? = nil
+    private var leadingContentViewController: DescribedViewController
     private lazy var leadingContainerView: ContainerView = .init()
 
     private lazy var separatorView: UIView = .init()
 
-    private var trailingContentViewController: ScreenViewController<TrailingScreenType>? = nil
+    private var trailingContentViewController: DescribedViewController
     private lazy var trailingContainerView: ContainerView = .init()
     
-    private var currentRatio: CGFloat {
-        didSet {
-            guard oldValue != currentRatio else {
-                return
-            }
-            
-            needsAnimatedLayout = true
-        }
-    }
-
-    private var currentSeparatorWidth: CGFloat {
-        didSet {
-            guard oldValue != currentSeparatorWidth else {
-                return
-            }
-
-            needsAnimatedLayout = true
-        }
-    }
-
     private var needsAnimatedLayout = false
 
-    required init(screen: ContainerScreen, viewRegistry: ViewRegistry) {
-        currentRatio = screen.ratio
-        currentSeparatorWidth = screen.separatorWidth
-        
-        super.init(screen: screen, viewRegistry: viewRegistry)
+    required init(screen: ContainerScreen) {
+        leadingContentViewController = DescribedViewController(screen: screen.leadingScreen)
+        trailingContentViewController = DescribedViewController(screen: screen.trailingScreen)
+        super.init(screen: screen)
     }
 
-    override public func screenDidChange(from previousScreen: ContainerScreen) {
+    override internal func screenDidChange(from previousScreen: ContainerScreen) {
+        if screen.ratio != previousScreen.ratio {
```

Comment:
Nice usage of `screenDidChange` to detect and set the `needsAnimatedLayout` flag. Didn't think about using this API for this.